src/shell/element/surface.rs

```diff
@@ -583,12 +531,18 @@ impl CosmicSurface {
                         .unwrap();
 
                     let current_server = attributes.current_server_state();
-                    if attributes.current.size == current_server.size {
+                    let mut guard = states.cached_state.get::<ToplevelCachedState>();
+                    if guard
+                        .current()
+                        .last_acked
+                        .as_ref()
+                        .is_some_and(|configure| configure.state.size == current_server.size)
+                    {
```

Comment:
Can't we simplify this whole thing with the new `with_committed_state` helper or am I misunderstanding how that works internally? Seems to me, that we don't need to manually check the ack'd serial.

Reply:
Yeah, I guess `with_committed_state()` would work here; though we also use `states.data_map`, in addition to the `states.cached_state`, to get the `current_server_state()`.

I guess it wouldn't really be a problem to take the `with_states` lock twice (but not at the same time) here? Though it might not simplify things much.